src/Commands/Sql/SqlDatabaseQueryService.cs

```diff
@@ -0,0 +1,50 @@
+using System.Threading.Tasks;
+using Azure.Identity;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Sql;
+using Azure.ResourceManager.Resources;
+using Microsoft.Data.SqlClient;
+using System.Collections.Generic;
+
+namespace AzureMcp.Commands.Sql;
+
+public sealed class SqlDatabaseQueryService() : ISqlDatabaseQueryService
+{
+    public async Task<object> ExecuteQueryAsync(string subscription, string serverName, string databaseName, string query)
+    {
+        var credential = new DefaultAzureCredential();
```

Comment:
This should extend from BaseAzureService, we want to use GetCredential() method in this class to fetch the credential.  (Getting credentials is an expensive operation.)

Similarly with the other services.

Reply:
Fixed